undos this "static" cast because current flex fails to compile result:

flex-2.6.4-1.x86_64

../pam_console.save/configfile.lex.c	2017-07-03 19:58:38.318088677 +0300
../pam_console/configfile.lex.c	2017-07-03 20:03:36.986144030 +0300
@@ -1089,7 +1089,7 @@
 
 extern int yylex (void);
 
-#define YY_DECL STATIC int yylex (void)
+#define YY_DECL int yylex (void)
 #endif /* !YY_DECL */

build log included:

bison -y -d -o configfile.tab.c -p _pc_yy configfile.y
configfile.y: warning: 1 shift/reduce conflict [-Wconflicts-sr]
sh ./sed-static configfile.tab.c
flex -Cr -oconfigfile.lex.c -P_pc_yy configfile.l
sh ./sed-static configfile.lex.c
/usr/lib64/ccache/x86_64-pld-linux-gcc -DHAVE_CONFIG_H -D_GNU_SOURCE -I. -I../..    -I../../libpam/include -I../../libpamc/include  -DLOCKDIR=\"/var/run/console\" -O2 -fwrapv -pipe -Wformat -Werror=format-security -gdwarf-4 -fno-debug-types-section -fvar-tracking-assignments -g2 -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector-strong --param=ssp-buffer-size=4 -fPIC -march=x86-64 -gdwarf-4 -fno-debug-types-section -fvar-tracking-assignments -g2  -W -Wall -Wbad-function-cast -Wcast-align -Wcast-qual -Wmissing-declarations -Wmissing-prototypes -Wpointer-arith -Wreturn-type -Wstrict-prototypes -Wwrite-strings -Winline -Wshadow -MT pam_console_apply-configfile.o -MD -MP -MF .deps/pam_console_apply-configfile.Tpo -c -o pam_console_apply-configfile.o `test -f 'configfile.c' || echo './'`configfile.c
In file included from configfile.c:56:0:
configfile.lex.c:23:15: error: static declaration of '_pc_yylex' follows non-static declaration
 #define yyleng _pc_yyleng
               ^
configfile.lex.c:1091:28: note: in expansion of macro 'yylex'
 
                            ^    
configfile.lex.c:1111:1: note: in expansion of macro 'YY_DECL'
  */
 ^~~    
configfile.lex.c:23:15: note: previous declaration of '_pc_yylex' was here
 #define yyleng _pc_yyleng
               ^
configfile.lex.c:1089:12: note: in expansion of macro 'yylex'
 
            ^    
In file included from configfile.c:57:0:
configfile.y:325:1: warning: 'do_yyerror' defined but not used [-Wunused-function]
 do_yyerror(const char *format, ...) {
 ^~~~~~~~~~
In file included from configfile.c:56:0:
configfile.lex.c:11:24: warning: '_pc_yy_scan_string' defined but not used [-Wunused-function]
 #define yy_scan_buffer _pc_yy_scan_buffer
                        ^
configfile.lex.c:2000:24: note: in expansion of macro 'yy_scan_string'
  */
                        ^             
In file included from configfile.c:56:0:
configfile.lex.c:1624:16: warning: 'input' defined but not used [-Wunused-function]
 #else
                ^    
configfile.lex.c:1581:17: warning: 'yyunput' defined but not used [-Wunused-function]
 
                 ^      
make[3]: *** [Makefile:698: pam_console_apply-configfile.o] Error 1
make[3]: Leaving directory '/home/users/glen/rpm/BUILD/x86_64-linux/Linux-PAM-1.3.0/modules/pam_console'

 
 /* Code executed at the beginning of each rule, after yytext and yyleng
--- Linux-PAM-1.3.0/modules/pam_console/sed-static~	2005-12-15 14:16:13.000000000 +0200
+++ Linux-PAM-1.3.0/modules/pam_console/sed-static	2017-07-03 20:03:11.824903776 +0300
@@ -13,7 +13,7 @@
 /^YY_BUFFER_STATE yy_scan_buffer/s/^/STATIC /
 /^YY_BUFFER_STATE yy_scan_string/s/^/STATIC /
 /^void yy_switch_to_buffer/s/^/STATIC /
-/define YY_DECL int yylex/s/YY_DECL /YY_DECL STATIC /
+#/define YY_DECL int yylex/s/YY_DECL /YY_DECL STATIC /
 /^int yyparse/s/^/STATIC /
 /^void yyrestart/s/^/STATIC /
 ' < $1 > sedtmp.$$ && mv sedtmp.$$ $1
